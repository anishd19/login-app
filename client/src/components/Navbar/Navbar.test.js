import React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import NavBar from "./Navbar";
import store from "../../store";

afterEach(cleanup);

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <NavBar />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders html elements", () => {
  const { getByText } = render(
    <Provider store={store}>
      <NavBar />
    </Provider>
  );

  expect(getByText(/Login App/i)).toBeTruthy();
  expect(getByText(/Log out/i)).toBeTruthy();
});

it("logout button click", () => {
  const { getByText } = render(
    <Provider store={store}>
      <NavBar />
    </Provider>
  );
  fireEvent.click(getByText(/Log out/i));
});
