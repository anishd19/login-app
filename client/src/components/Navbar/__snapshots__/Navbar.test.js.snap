// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="navbar"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
        href="#"
      >
        <div
          className="logo"
        >
          Login App
        </div>
      </a>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-end"
      >
        <a
          className="navbar-item"
          onClick={[Function]}
        >
          Log out
        </a>
      </div>
    </div>
  </div>
</div>
`;
